### DISCUSSION TITLE
about Router

### DISCUSSION THREAD

**Discussion by yjtwinpark** [Kitware/trame#215] [Q&A] [answered]
In vuetify example2,

with RouterViewLayout(server, "/bar/:id"):
    with vuetify.VCard():
         vuetify.VCardTitle("This is bar with ID '{{$route.params.id}}'"

localhost:8080/index.html#/bar/1, show VCardTitle as This is bar with ID '1'
I want to get this '1' as str value
How can I do that?

**Comment by jourdain** [comment, marked as answer]
I'm guessing you are asking to get it on the Python side?

Something like that should do the trick

```python
from trame.widgets import trame

state.active_id = None

@state.change("active_id")
def on_route_change(active_id, **kwargs):
   print(f"New route {active_id}")

with layout:
   trame.ClientStateChange(
      value="$route.params.id", 
      change="active_id = $route.params.id",
   )
```
